[feature branch merge] docs: restructure Getting Started section

qmonnet opened the pull request:
This PR aims at merging the [`feature/gsod-docs-reorg`](https://github.com/cilium/cilium/tree/feature/gsod-docs-reorg) feature branch. The feature branch was created because we did not want to risk leaving Cilium's docs in the middle of a major reorganisation when branching for 1.12. Branching is now finished, the reorganisation is progressing well and will be achieved before 1.13, so the feature branch is no longer necessary.

The feature branch currently contains the commits from these two PRs from @yoyo-go:

- https://github.com/cilium/cilium/pull/20175, rebased without conflict
- https://github.com/cilium/cilium/pull/20244, for which conflicts with the following PRs were addressed:

   - #19379 (File requirements-aks.rst split into two files)
   - #20369 (File host-services.rst removed)
   - #19760 (New files under gettingstarted/servicemesh/)
   - #20228 (New files under gettingstarted/clustermesh/)
   - #20439 (New files under gettingstarted/clustermesh/)
   - #20428 (New refs to update in troubleshooting_servicemesh.rst)
   - #20260 (One ref moved to troubleshooting_clustermesh.rst)

CODEOWNERS was also updated for both commits to reflect the updated paths.

Last commit adds a few fixes regarding to the `gs_guide` and `gs_install` link targets, to address some feedback from Joe.

cc @yoyo-go @xmulligan @lizrice

yoyo-go committed:
docs: Add Overview section

Note: This commit was first submitted via #20175 and merged to a feature
branch. This new version of the commit is similar, but rebased on the
master branch, and updates CODEOWNERS to update the renamed paths.

Signed-off-by: Yoyo Wu <yoyo19980720@126.com>
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

yoyo-go committed:
docs: Update Getting Started and Advanced Install. sections in a whole

Note: This commit was first submitted via #20244 and merged to a feature
branch. This new version of the commit is similar, but rebased on the
master branch, and addresses the conflicts with the following PRs:

    - #19379 (File requirements-aks.rst split into two files)
    - #20369 (File host-services.rst removed)
    - #19760 (New files under gettingstarted/servicemesh)
    - #20228 (New files under gettingstarted/clustermesh)
    - #20439 (New files under gettingstarted/clustermesh)
    - #20428 (New refs to update in troubleshooting_servicemesh.rst)
    - #20260 (One ref moved to troubleshooting_clustermesh.rst)

This commit also updates CODEOWNERS to reflect the updated paths, and
fixed a minor issue from the feature branch [0].

[0] https://github.com/cilium/cilium/pull/20681#discussion_r933352793

This is part of a reorganisation of Cilium's docs (GSoD project).

Signed-off-by: Yoyo Wu <yoyo19980720@126.com>
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

twpayne reviewed (approved):
👍 for IPAM (by running `make render-docs` and exploring).

joestringer commented on Documentation/concepts/observability/index.rst:
```diff
@@ -1,18 +0,0 @@
-   :glob:
-
-   intro
-   hubble-configuration
```
For the most part, this PR just moves files around so that seems fine. For some pages like this they appear to be deleted. I can't tell if this is intentional and another corresponding page was added elsewhere, or the reshuffle makes this page redundant, or something else.

qmonnet replied:
I don't think we intentionally deleted any of the pages.

`intro` and `hubble-configuration` are now `hubble_intro` and `hubble-configuration` (respectively) under `gettingstarted/`, and they are both listed in the root `Documentation/index.rst`.

Note that Sphinx would complain if some RST documents were not accessible from any of the index.rst, so we should be able to catch it from the CI.

joestringer commented on Documentation/index.rst:
```diff
@@ -62,6 +80,8 @@ get started and experiment with Cilium.
    operations/performance/index
    operations/troubleshooting
+   concepts/index
+   installation/microk8s
```
microk8s seems a bit random to just be added under operations here.

qmonnet replied:
I don't remember the details of that decision, to be honest. I think the idea was to deprecate it because nobody seemed to be using microk8s to run Cilium. @yoyo-go do you remember?

This is probably something we can address as a follow-up anyway.

joestringer commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -8,17 +8,17 @@
 This is the best installation method for most use cases.  For large
 environments (> 500 nodes) or if you want to run specific datapath modes, refer
-to the :ref:`k8s_install_advanced` guide.
+to the :ref:`gs_guide` guide.
```
Is this link replacement correct?

joestringer replied:
^Yes

joestringer commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -175,7 +175,7 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
        .. parsed-literal::
 
-          curl -LO \ |SCM_WEB|\/Documentation/gettingstarted/kind-config.yaml
+          curl -LO \ |SCM_WEB|\/Documentation/advancedinstallation/kind-config.yaml
```
Does `advancedinstallation` exist? At a glance it doesn't look like this is where the page ended up.

qmonnet replied:
Leftover from a previous naming scheme. I'll fix it, thanks for catching it!

joestringer commented on Documentation/installation/index.rst:
```diff
@@ -6,39 +6,64 @@
-Getting Started Guides
-======================
+Advanced Installation Guides
+============================
```
Probably `gs_guide` should point to the actual getting started guides, whereas `gs_install` should point to the installation guides?

qmonnet replied:
Right, there's even a few bits here and there that needed to be adjusted. I added a new commit (on top of those from the feature branch), would you mind taking a look please?

joestringer reviewed (approved):
Acknowledged. I understand this was all already reviewed & guided by you, Liz & Bill. 

We *might* still have some links pointing to some of these locations on external sites. We can set up redirects inside ReadTheDocs if necessary. I'm not sure whether we have a good way of detecting 404s when users hit the old pages though. Either way we probably won't find out until these docs changes hit `/stable` after 1.13 is released.

I glanced through anything that isn't just being moved from one place to another and noticed a few nits that might do with some closer attention.

qmonnet committed:
docs: Clean-up "gs_guide", "gs_install" link targets

“gs_guide” used to point to “Getting Started Guides”, but the section
title was recently changed into “Getting Started”. Let's rename the
label to “getting_started”, to avoid implying the ”guide” word, and
let's adjust the references accordingly by adding the missing word
wherever necessary. Let's also remove some outdated references to the
GSG related to Vagrant setups from dev_setup.rst.

“gs_install” is no longer used anywhere, let's remove it.

Reported-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

joestringer commented on Documentation/gettingstarted/k8s-install-default.rst:
```diff
@@ -18,7 +18,7 @@ distribution you are using. All state is stored using Kubernetes custom resource
 This is the best installation method for most use cases.  For large
 environments (> 500 nodes) or if you want to run specific datapath modes, refer
-to the :ref:`gs_guide` guide.
+to the :ref:`getting_started` guide.
```
Does this page describe how to run it specific datapath modes?

joestringer replied:
Ah I guess it's under the "Advanced Installation" section now. :+1:

qmonnet replied:
Ah true, I've restored (and renamed) the `gs_install` target for that case. Thanks!
